snuba/datasets/transactions.py

```diff
@@ -0,0 +1,87 @@
+from snuba.clickhouse import (
+    ColumnSet,
+    DateTime,
+    IPv4,
+    IPv6,
+    LowCardinality,
+    Materialized,
+    Nested,
+    Nullable,
+    String,
+    Tuple,
+    UInt,
+    UUID,
+    WithDefault,
+)
+from snuba.datasets import DataSet
+from snuba.datasets.schema import ReplacingMergeTreeSchema
+from snuba.datasets.transactions_processor import TransactionsMessageProcessor
+
+
+class TransactionsDataSet(DataSet):
+    def __init__(self):
+        columns = ColumnSet([
+            ('project_id', UInt(64)),
+            ('event_id', UUID()),
+            ('trace_id', UUID()),
+            ('span_id', UInt(64)),
+            ('transaction_name', String()),
+            ('transaction_hash', Materialized(
+                UInt(64),
+                'cityHash64(transaction_name)',
+            )),
+            ('transaction_op', LowCardinality(String())),
+            ('start_ts', DateTime()),
+            ('start_ms', UInt(16)),
+            ('finish_ts', DateTime()),
+            ('finish_ms', UInt(16)),
+            ('duration', Materialized(
+                UInt(32),
+                '((finish_ts - start_ts) * 1000) + (finish_ms - start_ms)',
+            )),
+            ('platform', LowCardinality(String())),
+            ('environment', Nullable(String())),
+            ('release', Nullable(String())),
+            ('dist', Nullable(String())),
+            ('ip_tup', Tuple([
+                ('v4', Nullable(IPv4())),
+                ('v6', Nullable(IPv6())),
+            ])),
```

Comment:
- If these are stored in separate files anyway, what is the benefit of storing this in a tuple?
- Can the members of the tuple be addressed without the tuple prefix? (E.g. are `ip_tup.1` and `v4` equivalent in a `SELECT` list?)

Reply:
_If these are stored in separate files anyway, what is the benefit of storing this in a tuple?_
I followed the schema we discussed two weeks ago. Unfortunately I could not find a final word on this type. Did we finally agree on using/avoiding Tuples ? @JTCunning ?

_Can the members of the tuple be addressed without the tuple prefix? (E.g. are ip_tup.1 and v4 equivalent in a SELECT list?)_
Don't know, will try

Reply:
I just had it as a tuple in the first draft. If it's easier to just define ipv4/ipv6 as separate columns, let's do it.